wfl/fit/mace.py

```diff
@@ -104,7 +103,16 @@ def fit(fitting_configs, mace_name, mace_fit_params, mace_fit_cmd, ref_property_
         return results
     
     run_dir.mkdir(parents=True, exist_ok=True)
-    
+   
+    # Copy training set xyz file to run_dir.
```

Comment:
I think it'd be nicer to have all of this logic in the same place, rather than having it split between here and the `_write_fitting_configs` function. For GAP it's all in this top level function, but in retrospect I think it's be better if it was all in a second smaller function, so if you do that here, I can refactor the GAP (and ACE?) functions. In fact, I almost think it's such a standard thing for fitting codes that we should write a single utility function that all fitting wrappers can call, but I can do that later.

Reply:
Can you simplify this bit of code to something like
```python
fitting_configs_scratch_filename = _prep_fitting_configs_file(mace_fit_params, fitting_configs)
```
rename `_write_fitting_configs` to `_prep_fitting_configs_file`, and add to it the call to `one_file()` and the `if ... else ...`. that are currently here? That way all the updating of `mace_fit_params` will happen in one place, and this modified function can return `None` if there's no scratch file, so the `unlink` later will still work.

Reply:
I think I have misunderstood your first comment, and I have removed that separate function. But now I understood your point. I'll rewrite a function named with that and push again.